test/evergreen.yml

```diff
@@ -749,21 +751,141 @@ tasks:
             set -o errexit
             set -o verbose
 
-            if [ "Windows_NT" = "$OS" ]; then
-              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
-            elif [ "$(uname -s)" == "Darwin" ]; then
-              # Avoid /usr/bin/python, at least on macOS: with System Integrity
-              # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
-              # doesn't find the WiredTiger library in the local tree.
-              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
-            else # Ubuntu
-              # Change directory to where the local installed 'wt' binary is located, 
-              # to avoid libtool generated 'wt' script from being selected by run.py,
-              # which invokes relink_command that tries to changing to a non-existed 
-              # /data/mci/<uniq> directory, as 'make' is done by a separate 'compile' task.
-              cd .libs
-              ${python_test_env_vars|} python ../test/suite/run.py -v 2 ${smp_command|} 2>&1
-            fi
+            # Only Windows and OS X variants are expected to run this task
+            #
+            # Avoid /usr/bin/python, at least on macOS: with System Integrity
+            # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
+            # doesn't find the WiredTiger library in the local tree.
+            ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+
+  # Break out Python unit tests into multiple buckets/tasks based on test name and runtime
+  # The test/suite/run.py script can work out test names by casting each command argument
+  # with "test_" prefix and "*.py" postfix. 
+  #
+  # One example: 
+  # "test/suite/run.py [ab]" will be translated to testing "test_a*.py" and "test_b*.py"
+
+  - name: unit-test-bucket00
+    depends_on:
+    - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            cd .libs
+            ${python_test_env_vars|} python ../test/suite/run.py [ab] -v 2 ${smp_command|} 2>&1
```

Comment:
It bothers me that there are missing letters in the buckets - if we add a test with one of those prefixes it wouldn't be run, and we wouldn't notice.

Reply:
I agree with Alex, we should maybe club all the missing ones together for any test we add in the future

Reply:
@agorrod I checked into the `test/suite/run.py` program and found this wild-casting [line](https://github.com/wiredtiger/wiredtiger/blob/develop/test/suite/run.py#L216) could help us achieve the wanted result by simply appending those missing letters to those existing letters in the existing 8 buckets. E.g. by adding missing letters 'f' and 'g' into bucket02 to let it run `${python_test_env_vars|} python ../test/suite/run.py [defg] -v 2 ${smp_command|} 2>&1`, the program will attempt to match `test_[defg]*.py`, and won't error out because 'd' and 'e' guarantee the existence of matched files. I pushed a commit just now which passed the patch build [here](https://evergreen.mongodb.com/version/5bfb30f1e3c331687f38bb11). Please let me know if you like this change.